apis/datadoghq/common/common.go

```diff
@@ -54,20 +51,11 @@ func GetDefaultReadinessProbe() *corev1.Probe {
 // GetImage builds the image string based on ImageConfig and the registry configuration.
 func GetImage(imageSpec *commonv1.AgentImageConfig, registry *string) string {
 	if defaulting.IsImageNameContainsTag(imageSpec.Name) {
-		// If the image name corresponds to a full URI we return it, otherwise
-		// we prefix name with passed registry or default one if former is empty/nil.
-		if len(strings.Split(imageSpec.Name, "/")) > 2 {
-			return imageSpec.Name
-		} else if registry != nil && *registry != "" {
-			return fmt.Sprintf("%s/%s", *registry, imageSpec.Name)
-		} else {
-			return fmt.Sprintf("%s/%s", DefaultImageRegistry, imageSpec.Name)
-		}
+		return imageSpec.Name
```

Comment:
This function is also called from code used in `controllers/datadogagent` which I believe is still called in operator v1.0.0 with V2 disabled. I'm not sure if this breaks something, just wanted to mention it just in case.

Reply:
This change basically reverts what was added in #652, so technically only change is added empty string check on line L59. Which I assume is needed since `registry` passed from [podtemplatespec](https://github.com/DataDog/datadog-operator/pull/664/files#diff-822a6227b8c46f98a0c6a5cb578d4439d84b09c84f332f28bfcfdd9a980d5aaaR152) won't be `nil`.